src/FixedSizeArray.jl

```diff
@@ -21,33 +21,49 @@ end
 const FixedSizeVector{T} = FixedSizeArray{T,1}
 const FixedSizeMatrix{T} = FixedSizeArray{T,2}
 
-function FixedSizeArray{T,N,V}(::UndefInitializer, size::NTuple{N,Int}) where {T,N,V}
-    new_fixed_size_array(V(undef, checked_dims(size))::V, size)
+function parent_type_with_default(::Type{<:(FixedSizeArray{E, N, T} where {N})}) where {E, T <: DenseVector{E}}
+    T
 end
-function FixedSizeArray{T,N,V}(::UndefInitializer, size::NTuple{N,Integer}) where {T,N,V}
-    ints = map(Int, size)::NTuple{N,Int}  # prevent infinite recursion
-    FixedSizeArray{T,N,V}(undef, ints)
+function parent_type_with_default(::Type{<:FixedSizeArray{E}}) where {E}
+    default_underlying_storage_type{E}
 end
-function FixedSizeArray{T,N,V}(::UndefInitializer, size::Vararg{Integer,N}) where {T,N,V}
-    FixedSizeArray{T,N,V}(undef, size)
+function parent_type_with_default(::Type{<:FixedSizeArray})
+    default_underlying_storage_type
```

Comment:
This is not concrete, right?

Reply:
That particular method shouldn't ever be called here, it's used in collect_as.

Reply:
All the `default_underlying_storage_type` methods were moved from collect_as.jl to this file in the first commit, to be shared between files.